frontend/src/features/children/components/childDetails.jsx

```diff
@@ -0,0 +1,114 @@
+import {useNavigate, useParams} from "react-router-dom";
+import {ChildQuery} from "../queries";
+import React, {useEffect, useState} from "react";
+import {APP_ROUTES} from "../../../core/routes";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
+import Stack from "@mui/material/Stack";
+import Button from "@mui/material/Button";
+import {EditableTitle} from "../../../shared";
+import {FieldsQuery} from "../../fields/queries";
+
+export const ChildDetails = () => {
+    const params = useParams();
+    const navigate = useNavigate();
+    const query = ChildQuery();
+    const child = query.useRead(params.id);
+    const updateMutation = query.useUpdate(params.id);
+    const deleteMutation = query.useDelete(params.id);
+    const [checked, setChecked] = useState([]);
+    const [name, setName] = useState();
+    const fieldsQuery = FieldsQuery();
+    const fields = fieldsQuery.useList();
+
+    const handleToggle = (event) => {
+        let value = event.target.tabIndex;
+        if (!checked.includes(value)) {
+            setChecked((preChecked) => [...preChecked, value]);
+        } else {
+            const currentIndex = checked.indexOf(value);
+            let newChecked = [...checked];
+            newChecked.splice(currentIndex, 1);
+            setChecked(newChecked);
+        }
+    };
+
+    const save = () => {
```

Comment:
consider using `try{}catch(){}` syntax
```  
const save = async () => {
    try{
      let child = {
        id: params.id,
        name: name,
        fields: checked,
      };
      await updateMutation.mutateAsync(child);
      setChecked(child.data.data.fields);
      setName(child.data.data.name);
      navigate(APP_ROUTES.CHILDREN.LIST);
    }
    catch(err){
      //handle err
    }
  };
 ```

In this case you could get rid of `useEffect` to handle side effects